src/artemis/experiment_plans/fast_grid_scan_plan.py

```diff
@@ -169,7 +221,11 @@ def gridscan_with_subscriptions(fgs_composite, detector, params):
     # the data were submitted to zocalo by the zocalo callback during the gridscan,
     # but results may not be ready, and need to be collected regardless.
     # it might not be ideal to block for this, see #327
-    xray_centre = subscriptions.zocalo_handler.wait_for_results(initial_xyz)
+    xray_centre, bbox_size = subscriptions.zocalo_handler.wait_for_results(initial_xyz)
+
+    if bbox_size is not None:
+        with TRACER.start_span("change_aperture"):
+            set_aperture_for_bbox_size(fgs_composite.aperture_scatterguard, bbox_size)
```

Comment:
As discussed, I don't think this is the idiomatic way of doing things as we're calling a function on the device during a plan. I think the better way is to create a signal on the device (that doesn't point to PVs) that we can write/read/wait on in the plan then the device will watch this signal and do the complex logic based on it. @callumforrester, does that sound correct to you?

Reply:
Why not just put the logic in `ApertureScatterguard.set`?, Then you can just go `yield from mv(aperture_device, aperture_device.LARGE)`, for example

Reply:
Oh yh, that's much cleaner!